### PR TITLE
Docker rewrite

### PR DESCRIPTION
**Pull request by SmithPeder**
## Docker rewrite

> Note: This is an "Endring av utforming" proposal, and all `.tex` files are left unchanged. 

<hr>

### Motivation

The old build process is a bit tedious, so we thought about moving away from `Jekyll`+`Makefile`+`Tempalte`+`Ruby` life, and over to the docker life.

The proposed change deletes almost all the build and template configuration from this repo. It replaces this with one Dockerfile, describing a docker image with all the necessary software to complete a full build. 
1. This will eliminate the need for large programs like `pandoc` and `latexmk` to be installed locally, which might now always work similarly for everyone.
2. This will eliminate the need for the `jekyll` and its associated ruby gems.
3. This will replace the `Makefile` build with an easy (in my opinion) `generate.sh` file, running inside the docker container.

<hr>

### New proposed structure 

As the `README` explains, the new structure boils down to three folders. One for **static** input (mainly html), one for **statutter**, and one for **public** output. The generate script will take convert the statutter to both PDF and HTML using and pipe it out to the public folder. This folder will then become the entrypoint for the website. The public folder now also acts as an archive, which holds all copies of statutter, one for each year. I have uploaded a couple of years.

### New proposed hosting

Again as the `README` explains, we propose to host this using `ZEIT` so it can be deployed in one command from the terminal. With this, we can avoid the whole Travis pipeline and Github pages.

[Qucik Demo](https://statutter-iqctrii32.now.sh/)

### CODE REVIEW

**Comment by kotlarz** [comment]
Demo link is broken FYI [fixed](https://statutter-iqctrii32.now.sh).

**Comment by SmithPeder** [comment]
Thanks for pointing that out, it's always that `https://` that gets you :/

File: generate.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+else
+   TIMESTAMP=$ASSEMBLY_DATE
+fi
+
+# Convert abakus LaTeX to html
+pandoc -f latex -t html \
+  --include-in-header=/static/head.html \
+  --include-before-body=/static/abakus.html \
+  --include-after-body=/static/end.html \
+  abakus-statutter/innhold.tex > "/public/index.html"
```
Comment (LudvigHz) [inline generate.sh:13]:
```suggestion
pandoc -f latex -o /public/index.html \
  --include-in-header=/static/head.html \
  --include-before-body=/static/abakus.html \
  --include-after-body=/static/end.html \
  abakus-statutter/innhold.tex
```

File: generate.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+# Move other static files to the public folder
+cp /static/main.css /static/abakule.jpg /public/
+
+# Convert abakus LaTeX to PDF
+pdflatex -halt-on-error abakus-statutter.tex
+pdflatex -halt-on-error abakus-statutter.tex
+pdflatex -pdf -halt-on-error -output-dir="/public" abakus-statutter.tex
```
Comment (LudvigHz) [inline generate.sh:21]: Why all these steps? At least the two identical ones.

File: generate.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+# Store a copy of the pdf in the archive
+mkdir -p /public/archive/
+cp /public/*.pdf /public/archive/$TIMESTAMP.pdf
```
Comment (LudvigHz) [inline generate.sh:25]: Using `*` here is a bit dangerous. Is it possible to use a defined file name.

File: generate.sh (hunk 2 above)
Comment (SmithPeder) [reply generate.sh:21]: https://www.systutorials.com/why-do-i-need-to-run-latex-bibtex-three-times-to-make-everything-look-good/

**Comment by SmithPeder** [comment]
I will close this since I made a fork, and it was annoying.

**Pull request closed without merging**